statsd/statsd.go

```diff
@@ -319,13 +340,29 @@ func newWithWriter(w io.WriteCloser, o *Options, writerName string) (*Client, er
 		tags:      o.tags,
 		telemetry: &statsdTelemetry{},
 	}
+
+	hasEntityID := false
 	// Inject values of DD_* environment variables as global tags.
 	for _, mapping := range ddEnvTagsMapping {
 		if value := os.Getenv(mapping.envName); value != "" {
+			if mapping.envName == ddEntityID {
+				hasEntityID = true
+			}
 			c.tags = append(c.tags, fmt.Sprintf("%s:%s", mapping.tagName, value))
 		}
 	}
 
+	// If DD_ENTITY_ID is set, we don't send a prefixed origin ID
+	// as dd.internal.entity_id is prioritized over dd.internal.prefixed_origin
+	if !hasEntityID {
+		// Sending a prefixed origin can be disabled by configuring DD_UDP_ORIGIN_DISABLED=true
+		if isDisabled := strings.ToLower(os.Getenv(udpOriginDisabled)); isDisabled != "true" {
```

Comment:
We could use https://pkg.go.dev/strconv#ParseBool here I think.

Also could we add an option for this (see the options.go file). And lastly, could we reverse the logic: instead of checking if it's disabled we check if it's enabled ?

Are we sure we want to enable this by default ? This would be a breaking change in behavior for customers (a new tag being sent that has an impact on the Agent behavior/performances).

Reply:
> We could use https://pkg.go.dev/strconv#ParseBool here I think.

> Also could we add an option for this (see the options.go file). And lastly, could we reverse the logic: instead of checking if it's disabled we check if it's enabled ?

Thanks! done!

> Are we sure we want to enable this by default ? This would be a breaking change in behavior for customers (a new tag being sent that has an impact on the Agent behavior/performances).

It's not enabled by default in the agent (see https://github.com/DataDog/datadog-agent/pull/10659). The agent doesn't even try to parse the field if the feature is not enabled. Having a single config in the agent will offer better UX.
cc @vickenty for visibility as this was your recommendation.